README.md

```diff
@@ -1 +1,84 @@
-# java-spiffe
\ No newline at end of file
+# JAVA-SPIFFE library
+
+## Overview
+
+The JAVA-SPIFFE library provides functionality to fetch SVIDs Bundles from a Workload API
+
+## SPIFFE Workload API ClientExample
+
+The `X509SvidFetcher` provides the method `registerListener` that allows a consumer to register a listener 
+to get the X509-SVIDS whenever the Workload API something new to send. 
+
+The channel is configured based on the Address (tcp or unix socket) and the OS detected.
+
+## Use
+```
+Fetcher<List<X509SVID>> svidFetcher = new X509SvidFetcher("/tmp/agent.sock");
+Consumer<List<X509SVID>> certificateUpdater;
+certificateUpdater = certs -> {
+    certs.forEach(svid -> {
+        System.out.println("Spiffe ID fetched: " + svid.getSpiffeId());
+    });
+};
+
+//Calling the WorkloadAPI to obtain the certificates
+svidFetcher.registerListener(certificateUpdater);
+```
+
+The `X509SvidFetcher` can be configured with a custom `RetryPolicy`. 
+
+By default it uses a `RetryPolicy` with the following parameters:
+
+```
+initialDelay = 1;
+maxDelay = 300;
+timeUnit = SECONDS;
+expBackoffBase = 2
+maxRetries = UNLIMITED_RETRIES;
+```
+
+### Running the example
+
+Required:
```

Comment:
Should we specify that JRE and JDK are required?
Can we enumerate the operating systems where you can run this example?

Reply:
Yes, thanks for the suggestion. Added that Java 8 is required and that the example was tested in Linux. Whether it runs on MacOS is still to be tested.